addons/flannel/template/base/install.sh

```diff
@@ -80,3 +91,83 @@ function flannel_weave_conflict() {
 function flannel_antrea_conflict() {
     ls /etc/cni/net.d/*antrea* >/dev/null 2>&1
 }
+
+function flannel_is_single_node() {
+    local nodecount=
+    nodecount=$(kubectl get nodes --no-headers | wc -l)
+    if [ "$nodecount" -eq 1 ]; then
+        return 0
+    else
+        return 1
+    fi
+}
+
+function weave_to_flannel() {
+    local dst="$DIR/kustomize/flannel"
+
+    echo "REMOVING WEAVE"
+    remove_weave
+
+    echo "UPDATING KUBEADM FOR FLANNEL"
+    flannel_kubeadm
+
+    echo "APPLYING FLANNEL"
+    kubectl -n kube-flannel apply -k "$dst/"
+
+    sleep 60
+    echo "RESTARTING KUBELET"
+    sudo systemctl stop kubelet
+    sudo iptables -t nat -F && sudo iptables -t mangle -F && sudo iptables -F && sudo iptables -X
+    sudo systemctl restart containerd
+    sudo systemctl start kubelet
+
+    sleep 60
```

Comment:
is there any way to wait for actual events here and for the subsequent sleeps? if not can we justify the sleep with a comment?

Reply:
Waiting for events in the containerd and kubelet restarts makes a lot of sense
same with the for-real flannel restart after this

I'm not sure how to generalize a wait for all the storageclasses easily though (and of course with "everything else" that gets restarted afterwards)